doc/article/en-US/angular1-to-aurelia.md

```diff
@@ -0,0 +1,526 @@
+		
+					scope.holeWidth = function () {
+						return Math.min(700, Math.max(350, _(scope.shop)
+							.reduce(function (p, c) {
+								return p + c.owned * c.digValue;
+							}, 0)
+						));
+					};
+		
+					scope.displayArray = [];
+		
+					scope.$watch('shop', function () {
+						scope.displayArray = _.values(scope.shop).reverse();
+					});
+		
+					(function tick(timestamp) {
+						angular.element($window).scrollTop($document.height());
+						animate(tick);
+					})();
+				}
+			};
+		}])
+	</source-code>
+</code-listing>
+
+Converting this directive to an Aurelia custom element requires us to remove the scope references, migrate from `link` and `controller` properties to lifecycle methods, and simplify our object. Finally, we'll have to use Aurelia's decorator functions to register our object and its configuration as a custom element instance.
+
+<code-listing heading="Aurelia Custom Components">
+	<source-code lang="ES 2015/2016">		export class Display {
+			constructor (window, document, animate) {
+				this.window = window;
+				this.document = document;
+				this.animate = animate;
+			}
+			
+			attached () {
+
+			}
+		}
+	</source-code>
+</code-listing>
+
+> Warning: Scope
+> Custom elements in Aurelia cannot access the outer scope. This means that all custom elements in Aurelia essentially behave as though they had the Angular property `scope: { ... }` - custom elements have an isolated scope.
+
+#### Converting an Angular directive to an Aurelia custom attribute
+
+Custom attributes are handled similarly to custom elements in Aurelia. Custom attributes also use an isolate scope - they cannot access the outer scope.
+
+#### Transclusion
+
+Aurelia provides the `content` tag to provide transclusion features. This is part of the view itself, instead of being part of the directive.
+
+####
+
+## [Quick Reference](aurelia-doc://section/4/version/1.0.0)
+
+#### $scope
+
+No equivalent exists in Aurelia.
+
+Instead of `$scope.property`, use `this.property` inside of a viewmodel object.
+
+Instead of `$scope.$watch()`, use a computed property - or, to trigger a function, use the BindingEngine.
+
+Instead of `$scope.$emit()` and `$scope.$broadcast()`, use `EventAggregator.publish()`. Likewise, instead of $scope.$on, use `EventAggregator.subscribe()`.
+
+#### $rootScope
+
+For signalling (using `$emit` and `$broadcast`, use Aurelia's event aggregator. For state that persists across viewmodels, use a separate service.
+
+#### $timeout
+
+No equivalent exists in Aurelia; it's not needed as Aurelia's change detection works differently to change detection in Angular. Aurelia will detect changes made from `setTimeout`, so use that.
+
+#### $cacheFactory
+
+Consider using Aurelia's cache plugin.
+
+#### $animate/$animateCss
+
+Use one of the 'aurelia-animate' plugins.
+
+#### $controller
+
+
+#### $document
+
+Use `window.document`.
+
+#### $filter
+
+Use a `ValueConverter`, providing your own `toView` and `fromView` methods.
```

Comment:
It is worth mentioning, that unlike Angular filters, Aurelia ValueConverters are two-way. The `toView` method would contain code taken from Angular filter. Explain, that `fromView` can be used to push changes from view back to the viewmodel.